api/src/main/java/com/mybooks/api/service/AuthorServiceImp.java

```diff
@@ -0,0 +1,51 @@
+package com.mybooks.api.service;
+
+import com.mybooks.api.exception.AuthorNotFoundException;
+import com.mybooks.api.model.Author;
+import com.mybooks.api.reposiotry.AuthorRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@AllArgsConstructor
+@Service
+public class AuthorServiceImp implements AuthorService {
+    private final AuthorRepository authorRepository;
+
+    @Override
+    public List<Author> getAllAuthors() {
+        return authorRepository.findAll();
+    }
+
+    @Override
+    public Author getAuthorById(String id) throws AuthorNotFoundException {
+        return authorRepository.findById(id).orElseThrow(() -> new AuthorNotFoundException(id));
```

Comment:
Maybe you can check if id is null or empty string.
Also test case should cover that as well.

Reply:
I implemete it to pass id in the url, for exampel `localhost:8080/book/author/1` so if the id is not added to link it would return all authors at this point. OR, do you prefer to pass `id` as a query parameter?

Reply:
Ok, np. I just need to mention that you should apply validations in service layer if its needed